### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by rgbkrk**
Rolled on through all of the documentation, clearing up language where I could. I'll keep going. My primary intention was to simplify a lot of the sections - introducing less concepts at once, trimming out the use of the word "just", cutting sections down, etc.

I was in the car, so I just kept reading and writing. Happy to break this up if you see fit. More still to come.

### CODE REVIEW

File: docs/introduction/concepts.md [hunk 1]
```diff
@@ -210,11 +204,9 @@ function fetchTodos() {
 }
 ```
 
-Now, this is of course doesn't abstract the side effect
-cleanly. But still you'd be pulling your requests directly
-in and dispatching an action as a result.
-
-To use a promise as a side effect do this instead:
+The primary problems with the above code is that it's coupled to the dispatcher
+and doesn't allow the dispatcher to reason about the any rejections in promises.
```
Comment (philpl) [inline docs/introduction/concepts.md]: `the any` - > `any` :) ?

File: docs/introduction/concepts.md [hunk 2]
```diff
@@ -244,21 +236,22 @@ creator using a thunk for the example above may look
...
+
+function thunkedTodos() {
   return dispatch => {
-    return fetch('/todo-list')
-      .then(res => res.json())
-      .then(todos => {
-        dispatch(addTodos(todos))
-        return todos
-      })
+    dispatch(fetchTodos)
```
Comment (philpl) [inline docs/introduction/concepts.md]: Lol I could've thought of that :D that's much easier to read

**Comment by philpl** [comment]
Thanks, this is awesome :) I like your changes a lot. Cleans it up, makes it easier to read. Just found a single typo while skimming 🚢

File: docs/introduction/concepts.md (hunk 1 above)
Comment (rgbkrk) [inline docs/introduction/concepts.md]: Gah! Room for more PRs!

File: docs/introduction/concepts.md (hunk 2 above)
Comment (rgbkrk) [inline docs/introduction/concepts.md]: Of note: I had no idea what to name this.

**Pull request merged**